WordPress/Classes/ViewRelated/Blog/DiscussionSettingsViewController.swift

```diff
@@ -83,8 +83,8 @@ public class DiscussionSettingsViewController : UITableViewController
         }
         
         let service = BlogService(managedObjectContext: settings.managedObjectContext)
-        service.updateSettingsForBlog(settings.blog,
-            success: nil,
+        service.updateSettingsForBlog(blog,
+            success: { _ in },
```

Comment:
Technically, no it's not :wink: It was before this PR, and the code will work with a nil closure. I considered the alternative as well, but I think as we write more swift it makes sense to move towards nonnull closures. 

If there's a callback, I think it's safe to assume that it will be provided most of the time, otherwise it wouldn't be there. I also think it's reasonable to put the burden on the caller instead of the callee: I'd much rather have `completion: { _ in }` than `if (completion) { completion() }`

At the same time, only one caller `updateSettingsForBlog` provides a `success` block, and I believe it's a wrong implementation (it uses success to call reloadData, but updateSettingsForBlog won't change anything). So effectively we have no use case for a success block here. But I'll stop before I go on full rant mode against the success/failure pattern.

In any case, I agree that this change is out of scope for adding the nullability annotations :smile: